.github/workflows/nightly.yml

```
@@ -0,0 +1,143 @@
+name: Nightly
+on:
+  # FIXME: Change to run on a schedule before merging
+  #schedule:
+  #  - cron: "0 1 * * *"
+  pull_request:
+    branches:
+      - '**'
+jobs:
+  GENERATE_MATRIX:
+    name: Generate Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.set-matrix.outputs.branches }}
+      asan-matrix: ${{ steps.set-matrix.outputs.asan-matrix }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get current date for cache
+        id: date
+        run: echo "::set-output name=date::$(date -Iseconds)"
+      - name: Grab the commit mapping
+        uses: actions/cache@v3
+        with:
+          path: branch-commit-cache.json
+          key: nightly-${{ steps.date.outputs.date }}
+          restore-keys: |
+            nightly-
+      - name: Generate Matrix
+        id: set-matrix
+        run: php .github/nightly_matrix.php
```

```suggestion
        run: php .github/nightly_matrix.php "${{ github.event_name }}" "${{ github.run_id }}" "${{ github.run_attempt }}"
```

This will pass:
- The event_name (either `schedule`, `push` or `workflow_dispatch`).
- The run_id.
- The run_attempt (“Retry counter”).

You can use this to bypass the cached data for manual attempts (workflow_dispatch) or retries (run_attempt > 1). This might come in handy if the cache gets completely borked for whatever reason.